main.go

```diff
@@ -92,31 +111,72 @@ func main() {
 		log.Fatalf("Failed processing the swagger spec: %s", err)
 	}
 
-	if err := models.Generate(*goPackageName, swaggerSpec); err != nil {
+	if err := models.Generate(goPackagePath, swaggerSpec); err != nil {
 		log.Fatalf("Error generating models: %s", err)
 	}
 
-	if err := server.Generate(*goPackageName, swaggerSpec); err != nil {
+	if err := server.Generate(*goPackageName, goPackagePath, swaggerSpec); err != nil {
 		log.Fatalf("Failed to generate server: %s", err)
 	}
 
-	if err := goclient.Generate(*goPackageName, swaggerSpec); err != nil {
+	if err := goclient.Generate(*goPackageName, goPackagePath, swaggerSpec); err != nil {
 		log.Fatalf("Failed generating go client %s", err)
 	}
 
 	if err := jsclient.Generate(*jsModulePath, swaggerSpec); err != nil {
 		log.Fatalf("Failed generating js client %s", err)
 	}
 
-	middlewareGenerator := swagger.Generator{PackageName: *goPackageName}
+	middlewareGenerator := swagger.Generator{PackagePath: goPackagePath}
 	middlewareGenerator.Write(hardcoded.MustAsset("../_hardcoded/middleware.go"))
 	if err := middlewareGenerator.WriteFile("server/middleware.go"); err != nil {
 		log.Fatalf("Failed to copy middleware.go: %s", err)
 	}
 
-	doerGenerator := swagger.Generator{PackageName: *goPackageName}
+	doerGenerator := swagger.Generator{PackagePath: goPackagePath}
 	doerGenerator.Write(hardcoded.MustAsset("../_hardcoded/doer.go"))
 	if err := doerGenerator.WriteFile("client/doer.go"); err != nil {
 		log.Fatalf("Failed to copy doer.go: %s", err)
 	}
 }
+
+func getModulePackagePath(outputPath string) string {
+	pwd, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("Error getting current directory: %s", err.Error())
+	}
+	goPath := os.Getenv("GOPATH")
+	if goPath == "" {
+		log.Fatalf("GOPATH must be set")
+	}
+	if outputPath == "" {
+		log.Fatal("output-path is required")
+	}
+	goSrcPath := fmt.Sprintf("%v%v", goPath, "/src/")
+	return path.Join(strings.TrimPrefix(pwd, goSrcPath), outputPath)
+}
+
+func getModulePackageName(packagePath string, modFile *os.File) string {
+	// read first line of module file
+	r := bufio.NewReader(modFile)
+	b, _, err := r.ReadLine()
+	if err != nil {
+		log.Fatalf("Error checking module name: %s", err.Error())
+	}
+
+	// parse module version
+	moduleName := strings.TrimPrefix(string(b), "module")
+	moduleName = strings.TrimSpace(moduleName)
+	modulePath, pathMajor, ok := module.SplitPathVersion(moduleName)
+	if !ok {
+		log.Fatalf("invalid module path %q", modulePath)
+	}
+	pseudoMajor := module.PathMajorPrefix(pathMajor)
+
+	// add module version to package path
+	if pseudoMajor != "" {
+		suffix := strings.TrimPrefix(packagePath, modulePath)
+		return fmt.Sprintf("%v/%v%v", modulePath, pseudoMajor, suffix)
+	}
+	return modulePath
```

Comment:
haven't had a chance to dig into this but am confused by it, probably some comments about what the inputs/outputs are would be helpful

Reply:
let me know if the new function comment helps d4f02d2